src/Progress/Snapshot.php

```diff
@@ -4,26 +4,95 @@
 namespace BristolSU\Support\Progress;
 
 
+use BristolSU\Support\Activity\Activity;
 use BristolSU\Support\ActivityInstance\ActivityInstance;
 use BristolSU\Support\ActivityInstance\Contracts\ActivityInstanceRepository;
 use BristolSU\Support\ModuleInstance\Facade\ModuleInstanceEvaluator;
+use BristolSU\Support\Progress\ProgressUpdateRepository;
 use Carbon\Carbon;
+use Illuminate\Support\Facades\Hash;
 
 class Snapshot
 {
 
+    private ProgressUpdateRepository $ProgressUpdateRepository;
+
+    public function __construct()
+    {
+        $this->ProgressUpdateRepository = new ProgressUpdateRepository();
```

Comment:
This needs to go in the construct function (and typehint the interface), otherwise we can't benefit from dependency injection and will be tied to that repository.

If tests fail because of this (should only be in the `SnapshotTest` class), just mock the repository and pass it to any new instances of this class instead of a repository implementation
`new Snapshot($this->prophesize(\BristolSU\Support\Progress\Contracts\ProgresUpdateContract::class)->reveal());`